posts/ja/2023-02-13-22.0.0.13.adoc

```diff
@@ -0,0 +1,202 @@
+
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/23403[[HTTP/2ストリームが例外で終了したときに、断続的にサーバーの停止に失敗する] [HTTP/2ストリームが例外で終了したときに、断続的にサーバーの停止に失敗する]
++
+HTTP/2接続で例外が発生すると、スレッドがハングアップする問題が断続的に発生することがあります。この結果、サーバー停止時にquiesce警告が発行されます。
++
+この問題は解決され、HTTP/2のストリームと接続、および管理するスレッドは、例外状態でもエラーなしで停止するようになりました。
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/23326[LibertyのデフォルトのHttpAuthenticationMechanismsは、HttpMessageContext.responseUnauthorizedを呼び出さない]
++
+`HttpMessageContextWrapper` は `responseUnauthorized` メソッドをオーバーライドする必要があります。バグのため、Liberty の `HttpAuthenticationMechanisms` は `HttpMessageContext.responseUnauthorized` を呼び出さないので、ユーザーが例えばヘッダーを追加してレスポンスをエンリッチ化することができませんでした。
++
+この問題は解決され、デフォルトの `HttpAuthenticationMechanisms` は認証されていないリクエストに対して `HttpMessageContext.responseUnauthorized` を適切に呼び出すようになりました。
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/23146[JspFactory.getDefaultFactory().getEngineInfo().getSpecificationVersion()不正なバージョンを返す]
++
+Pages (旧 JSP) アプリケーションが `JspFactory.getDefaultFactory().getEngineInfo().getSpecificationVersion()` を呼び出すと、間違ったバージョンが返されることがあります。この値は、正しい仕様のバージョンではなく、`2.1` を返すようにコード化されています。 
++
+この問題は解決され、正しいバージョンが返されるようになりました。 例えば、`jsp-2.2` は `2.2` を返し、`pages-3.0` は `3.0` を返します。
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/22405[OidcClientImplがSecurityServiceへの依存関係を正しく宣言していない]
++
+フォームベースのログアウトURLである `ibm_security_logout` へのトラフィックを遮断する `ServletFilter` から `request.logout()` を呼び出すと、以下のNPEが発生する可能性があります。:
+[source]
+----
+java.lang.NullPointerException
+Stack Dump = java.lang.NullPointerException
+        at com.ibm.ws.security.openidconnect.client.internal.OidcClientImpl.authenticateSubject(OidcClientImpl.java:749)
+        at com.ibm.ws.security.openidconnect.client.internal.OidcClientImpl.handleOidcCookie(OidcClientImpl.java:722)
+        at com.ibm.ws.security.openidconnect.client.internal.OidcClientImpl.logout(OidcClientImpl.java:663)
+        at com.ibm.ws.webcontainer.security.AuthenticateApi.logoutUnprotectedResourceServiceRef(AuthenticateApi.java:244)
+        at com.ibm.ws.webcontainer.security.AuthenticateApi.logout(AuthenticateApi.java:189)
+        at com.ibm.ws.webcontainer.security.AuthenticateApi.logoutServlet30(AuthenticateApi.java:627)
+        at com.ibm.ws.webcontainer.security.WebAppSecurityCollaboratorImpl.logout(WebAppSecurityCollaboratorImpl.java:1212)
+        at com.ibm.ws.webcontainer.srt.SRTServletRequest.logout(SRTServletRequest.java:3956)
+        at javax.servlet.http.HttpServletRequestWrapper.logout(HttpServletRequestWrapper.java:376)
+        at com.ibm.bpm.servlet.filters.GenericSecurityServletFilter.doFilter(GenericSecurityServletFilter.java:327)
+        at com.ibm.ws.webcontainer.filter.FilterInstanceWrapper.doFilter(FilterInstanceWrapper.java:201)
+----
++
+The issue has been resolved by declaring `OidcClientImpl` dependent on `SecurityService` and the NPE no longer occurs.
```

Comment:
Does this need to be translated?

Reply:
It is translated. Please remove (I tried but I may not have authority to)